### ISSUE TITLE
[BUG] ColorConversionNeededError in v2.2.1

### ISSUE THREAD

**Issue by tofublock** [paperless-ngx/paperless-ngx#5214]
### Description

Similar to #4702:
Uploading a specific PDF, I receive
`XYZ.pdf: Error occurred while consuming document XYZ.pdf: ColorConversionNeededError: The input PDF has an unusual color space. Use --color-conversion-strategy to convert to a common color space
such as RGB, or use --output-type pdf to skip PDF/A conversion
and retain the original color space.`

In `docker-compose.env`: Setting `PAPERLESS_OCR_COLOR_CONVERSION_STRATEGY=RGB` or `PAPERLESS_OCR_COLOR_CONVERSION_STRATEGY=LeaveColorUnchanged` does not help, but setting `PAPERLESS_OCR_COLOR_CONVERSION_STRATEGY=Gray` works.

Is it possible that the parameters aren't passed on to OCRmyPDF correctly?

### Steps to reproduce

Error appears on upload of one specific PDF1.3 document. I don't see anything particularly interesting about the PDF.

### Webserver logs

```bash
[2024-01-02 19:13:51,288] [ERROR] [celery.app.trace] Task documents.tasks.consume_file[67d07ac5-63fa-4db3-95f8-bb59b577433c] raised unexpected: ConsumerError('XYZ.pdf: Error occurred while consuming document XYZ.pdf: ColorConversionNeededError: The input PDF has an unusual color space. Use\n--color-conversion-strategy to convert to a common color space\nsuch as RGB, or use --output-type pdf to skip PDF/A conversion\nand retain the original color space.\n')
Traceback (most recent call last):
  File "/usr/src/paperless/src/paperless_tesseract/parsers.py", line 330, in parse
    ocrmypdf.ocr(**args)
  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/api.py", line 375, in ocr
    return run_pipeline(options=options, plugin_manager=plugin_manager)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/_pipelines/ocr.py", line 225, in run_pipeline
    return _run_pipeline(options, plugin_manager)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/_pipelines/ocr.py", line 192, in _run_pipeline
    optimize_messages = exec_concurrent(context, executor)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/_pipelines/ocr.py", line 148, in exec_concurrent
    pdf, messages = postprocess(pdf, context, executor)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/_pipelines/_common.py", line 420, in postprocess
    pdf_out = convert_to_pdfa(pdf_out, ps_stub_out, context)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/_pipeline.py", line 814, in convert_to_pdfa
    context.plugin_manager.hook.generate_pdfa(
  File "/usr/local/lib/python3.11/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pluggy/_callers.py", line 113, in _multicall
    raise exception.with_traceback(exception.__traceback__)
  File "/usr/local/lib/python3.11/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/builtin_plugins/ghostscript.py", line 128, in generate_pdfa
    ghostscript.generate_pdfa(
  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/_exec/ghostscript.py", line 296, in generate_pdfa
    raise ColorConversionNeededError()
ocrmypdf.exceptions.ColorConversionNeededError: The input PDF has an unusual color space. Use
--color-conversion-strategy to convert to a common color space
such as RGB, or use --output-type pdf to skip PDF/A conversion
and retain the original color space.


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 349, in main_wrap
    raise exc_info[1]
  File "/usr/src/paperless/src/documents/consumer.py", line 446, in try_consume_file
    document_parser.parse(self.path, mime_type, self.filename)
  File "/usr/src/paperless/src/paperless_tesseract/parsers.py", line 397, in parse
    raise ParseError(f"{e.__class__.__name__}: {e!s}") from e
documents.parsers.ParseError: ColorConversionNeededError: The input PDF has an unusual color space. Use
--color-conversion-strategy to convert to a common color space
such as RGB, or use --output-type pdf to skip PDF/A conversion
and retain the original color space.


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/celery/app/trace.py", line 477, in trace_task
    R = retval = fun(*args, **kwargs)
                 ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/celery/app/trace.py", line 760, in __protected_call__
    return self.run(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/paperless/src/documents/tasks.py", line 167, in consume_file
    document = Consumer().try_consume_file(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/paperless/src/documents/consumer.py", line 474, in try_consume_file
    self._fail(
  File "/usr/src/paperless/src/documents/consumer.py", line 115, in _fail
    raise ConsumerError(f"{self.filename}: {log_message or message}") from exception
documents.consumer.ConsumerError: XYZ.pdf: Error occurred while consuming document XYZ.pdf: ColorConversionNeededError: The input PDF has an unusual color space. Use
--color-conversion-strategy to convert to a common color space
such as RGB, or use --output-type pdf to skip PDF/A conversion
and retain the original color space.
```

### Paperless-ngx version

2.2.1

### Host OS

Debian x86_64

### Installation method

Docker - official image

### Browser

Firefox

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

**Comment by shamoon** [comment]
If you look at the full logs (before the part you included) you can see the exact options that are being sent to ocrmypdf. Given how simple this is (see https://github.com/paperless-ngx/paperless-ngx/pull/4709/files ) I don’t think we’re passing them incorrectly, more likely is that they aren’t behaving as you expected. To be fair, I don’t find it totally clear but as noted this isn’t really up to us, it’s directly from ocrmypdf

Basically, look at how it’s calling ocrmypdf and try it yourself, if that recreates it, the issue isn’t with paperless

**Comment by stumpylog** [comment]
Yeah, it's dead simple to pass on to ocrmypdf.  Most likely, it's not reporting a ghostscript error quite right and just reports this one instead.  It would explain why Gray works, ghostscript can handle the file with that setting, but not others.

**Comment by tofublock** [comment]
Okay, thanks for taking a look and the explanation; that makes sense. Curious error upstream, though.
